graph_dfs_debug/graph.py

```diff
@@ -2,39 +2,65 @@
 Simple graph implementation compatible with BokehGraph class.
 """
 class Vertex:
+    # so why is component default value -1?
     def __init__(self, label, component=-1):
         self.label = str(label)
         self.component = component
 
     def __repr__(self):
         return 'Vertex: ' + self.label
 
-    """Trying to make this Graph class work..."""
+# v = Vertex("me")
+# print(v.component)
+# print(v.label)
+# print(v)
+
+    #Trying to make this Graph class work..."""
 class Graph:
     def __init__(self):
+        # so again graph is just a dictionary of sets
         self.vertices = {}
         self.components = 0
 
+    def get_vertex_label(self, vertex):
+        pass
+
     def add_vertex(self, vertex, edges=()):
-        self.vertices[vertex] = set(edges)
+        self.vertices[Vertex(vertex)] = set(edges)
```

Comment:
We don't need to instantiate `Vertex` here since we pass a `Vertex` instance when we call `add_vertex`.